### PR TITLE
WSL full support

### LINKED ISSUE #749: docker-18.06.0-ce binaries for Windows
As with #537, we'll have manually include the docker-18.06.0-ce binaries for Windows in the next release.

### LINKED ISSUE #851: SMB3 support for Windows 10
Looks like we may be able to use SMB3 with the latest boot2docker (v18.09.0). It switched to a new Tiny Core / Linux kernel version and should now support SMB3.

https://github.com/boot2docker/boot2docker/blob/9ca8f5bce3377787cc20a815f70b497a195a5760/files/kernel-config.d/cifs#L5-L9

### LINKED ISSUE #865: Revise WSL use case on Windows (installation and usage)
- [ ] Pin Docker client version (currently installs the latest)
- [ ] Test and make WSL usable with both VirtualBox as well as Docker Desktop
- [ ] Change installation docs (video still points to test.docksal.io)

### LINKED ISSUE #936: Docksal installation should not quietly replace docker-compose on Linux
Currently Docksal installation on Linux puts docker-compose to `/usr/local/bin`, potentially overwriting the version people had. We need to either put it to our own folder as we do on other OS, or at least issue a notification.

### LINKED ISSUE #949: vm start does not work as part of fin up
If you're in a project that is not "up" and the Virtualbox machine is not running and you run `fin up`, the VM never starts correctly on the first try.   I always have to run `fin vm restart` when it errors out.   this happens on all my projects

```
Kanopi-MBP:mises.org paulsheldrake$ fin up
It looks like 'docksal' docker machine is not running.
Run 'fin vm start' to start it now? [y/n]: y
Starting "docksal"...
(docksal) Waiting for an IP...
Machine "docksal" was started.
Waiting for SSH to be available...
Detecting the provisioner...
Started machines may have new IP addresses. You may need to re-run the `docker-machine env` command.
Enabling automatic *.docksal DNS resolver...
Clearing DNS cache...
Password:
Configuring NFS shares...
NFS shares are already configured
Mounting NFS shares...
/usr/local/sbin/rpcbind is already running
3552
/usr/local/sbin/rpc.statd is already running
3554
Starting nfs client utilities.
Mounting local /Users/paulsheldrake/docksal to /Users/paulsheldrake/docksal
Importing ssh keys...
Key 'id_rsa' already loaded in the agent. Skipping.
 ERROR:  Your Docker client is unable to talk to the Docker server.
         WHAT TO DO?
         1. Try fin vm restart.
         2. If #1 did not help, run fin update to update.
         3. Check errors you see in troubleshooting doc https://docs.docksal.io/troubleshooting/common-issues/
Kanopi-MBP:mises.org paulsheldrake$ fin vm restart
Stopping "docksal"...
Machine "docksal" was stopped.
Starting "docksal"...
(docksal) Check network to re-create if needed...
(docksal) Waiting for an IP...
Machine "docksal" was started.
Waiting for SSH to be available...
Detecting the provisioner...
Started machines may have new IP addresses. You may need to re-run the `docker-machine env` command.
Enabling automatic *.docksal DNS resolver...
Clearing DNS cache...
Configuring NFS shares...
NFS shares are already configured
Mounting NFS shares...
Starting nfs client utilities.
Mounting local /Users/paulsheldrake/docksal to /Users/paulsheldrake/docksal
Importing ssh keys...
Identity added: id_rsa (id_rsa)
Kanopi-MBP:mises.org paulsheldrake$
```

<details>
  <summary>fin config output</summary>

```
---------------------
COMPOSE_PROJECT_NAME_SAFE: misesorg
COMPOSE_FILE:
/Users/paulsheldrake/.docksal/stacks/volumes-bind.yml
/Users/paulsheldrake/docksal/kanopi_docksal/mises/mises.org/.docksal/docksal.yml
ENV_FILE:
/Users/paulsheldrake/docksal/kanopi_docksal/mises/mises.org/.docksal/docksal.env
/Users/paulsheldrake/docksal/kanopi_docksal/mises/mises.org/.docksal/docksal-local.env

PROJECT_ROOT: /Users/paulsheldrake/docksal/kanopi_docksal/mises/mises.org
DOCROOT: .
VIRTUAL_HOST: mises.docksal
VIRTUAL_HOST_ALIASES: *.mises.docksal
IP: 192.168.64.100
MYSQL:

Docker Compose configuration
---------------------
services:
  blackfire:
    environment:
      BLACKFIRE_SERVER_ID: aa4ac*****92704
      BLACKFIRE_SERVER_TOKEN: b2876*****0ac71
    image: blackfire/blackfire
  cli:
    dns:
    - 192.168.64.100
    - 8.8.8.8
    environment:
      BLACKFIRE_CLIENT_ID: null
      BLACKFIRE_CLIENT_TOKEN: null
      DOCROOT: .
      GIT_USER_EMAIL: paul.sheldrake@gmail.com
      GIT_USER_NAME: Paul Sheldrake
      HOST_GID: '20'
      HOST_UID: '501'
      LANG: en_US.UTF-8
      LANGUAGE: en_US.UTF-8
      LC_ALL: en_US.UTF-8
      SECRET_ACAPI_EMAIL: null
      SECRET_ACAPI_KEY: null
      SECRET_PLATFORMSH_CLI_TOKEN: null
      SECRET_SSH_PRIVATE_KEY: null
      SECRET_TERMINUS_TOKEN: null
      VIRTUAL_HOST: mises.docksal
      XDEBUG_CONFIG: remote_connect_back=0 remote_host=192.168.64.1
      XDEBUG_ENABLED: '0'
    hostname: cli
    image: docksal/cli:1.3-php5
    volumes:
    - docksal_ssh_agent:/.ssh-agent:ro
    - /Users/paulsheldrake/docksal/kanopi_docksal/mises/mises.org/.docksal/etc/drush:/etc/drush:ro
    - cli_home:/home/docker:rw
    - project_root:/var/www:rw,nocopy
  db:
    dns:
    - 192.168.64.100
    - 8.8.8.8
    environment:
      MYSQL_DATABASE: default
      MYSQL_PASSWORD: user
      MYSQL_ROOT_PASSWORD: root
      MYSQL_USER: user
    hostname: db
    image: docksal/db:1.1-mysql-5.7
    ports:
    - 3306/tcp
    volumes:
    - db_data:/var/lib/mysql:rw
    - project_root:/var/www:ro,nocopy
  mail:
    environment:
      MH_API_BIND_ADDR: 0.0.0.0:80
      MH_UI_BIND_ADDR: 0.0.0.0:80
    expose:
    - '80'
    hostname: mail
    image: mailhog/mailhog
    labels:
      io.docksal.virtual-host: webmail.mises.docksal
    user: root
  memcached:
    command:
    - -m
    - '128'
    dns:
    - 192.168.64.100
    - 8.8.8.8
    hostname: memcached
    image: memcached:1.4-alpine
  web:
    depends_on:
      cli:
        condition: service_started
    dns:
    - 192.168.64.100
    - 8.8.8.8
    environment:
      APACHE_BASIC_AUTH_PASS: null
      APACHE_BASIC_AUTH_USER: null
      APACHE_DOCUMENTROOT: /var/www/.
      APACHE_FILE_PROXY: https://mises.org
      BLACKFIRE_CLIENT_ID: null
      BLACKFIRE_CLIENT_TOKEN: null
    hostname: web
    image: docksal/web:2.1-apache2.4
    labels:
      io.docksal.cert-name: none
      io.docksal.permanent: "false"
      io.docksal.project-root: /Users/paulsheldrake/docksal/kanopi_docksal/mises/mises.org
      io.docksal.virtual-host: mises.docksal,*.mises.docksal,mises.docksal.*
    volumes:
    - /Users/paulsheldrake/docksal/kanopi_docksal/mises/mises.org/.docksal/etc/apache/host.conf:/usr/local/apache2/conf/extra/includes/host.conf:ro
    - project_root:/var/www:ro,nocopy
version: '2.1'
volumes:
  cli_home: {}
  db_data: {}
  docksal_ssh_agent:
    external: true
    name: docksal_ssh_agent
  project_root:
    driver: local
    driver_opts:
      device: /Users/paulsheldrake/docksal/kanopi_docksal/mises/mises.org
      o: bind
      type: none

---------------------
```

</details>

**Output of `fin sysinfo`:**

<details>
  <summary>fin sysinfo output</summary>

```
███  OS
Darwin Mac OS X 10.12.6
Darwin Kanopi-MBP.hitronhub.home 16.7.0 Darwin Kernel Version 16.7.0: Mon Nov 13 21:56:25 PST 2017; root:xnu-3789.72.11~1/RELEASE_X86_64 x86_64

███  ENVIRONMENT
MODE : VirtualBox VM
DOCKER_HOST : tcp://192.168.64.100:2376
DOCKSAL_NFS_PATH : /Users/paulsheldrake/docksal
NFS EXPORTS:
# VAGRANT-BEGIN: 501 19ed865d-28dd-4d57-931b-0fc57b856b49
"/Users/paulsheldrake/.vagrant.d/cache/geerlingguy/ubuntu1604" 192.168.88.88 -alldirs -mapall=501:20
"/Users/paulsheldrake/Documents/work/kanopi/pnw/pnw_drupalvm" 192.168.88.88 -alldirs -mapall=501:20
# VAGRANT-END: 501 19ed865d-28dd-4d57-931b-0fc57b856b49
# VAGRANT-BEGIN: 501 29e3522b-b1c7-4494-8aff-83deff88c628
"/Users/paulsheldrake/.vagrant.d/cache/geerlingguy/ubuntu1604" 192.168.33.88 -alldirs -mapall=501:20
"/Users/paulsheldrake/Documents/work/kanopi/zopfli_vm" 192.168.33.88 -alldirs -mapall=501:20
# VAGRANT-END: 501 29e3522b-b1c7-4494-8aff-83deff88c628
# VAGRANT-BEGIN: 501 409fe48b-b565-4a78-a5b4-5d3e4b7591cd
"/Users/paulsheldrake/.vagrant.d/cache/geerlingguy/ubuntu1604" 192.168.88.88 -alldirs -mapall=501:20
"/Users/paulsheldrake/Documents/work/kanopi/ico/drupalvm_ico" 192.168.88.88 -alldirs -mapall=501:20
"/Users/paulsheldrake/Documents/work/kanopi/ico/ico-rally" 192.168.88.88 -alldirs -mapall=501:20
# VAGRANT-END: 501 409fe48b-b565-4a78-a5b4-5d3e4b7591cd
# VAGRANT-BEGIN: 501 e48f4e7d-f212-4fb7-b14f-5d78fb7df132
"/Users/paulsheldrake/.vagrant.d/cache/geerlingguy/centos6" 10.20.1.3 -alldirs -mapall=501:20
"/Users/paulsheldrake/Documents/work/kanopi/zuckerman/zuckerman" "/Users/paulsheldrake/Documents/work/kanopi/zuckerman/zuckerman/htdocs" "/Users/paulsheldrake/Documents/work/kanopi/zuckerman/zuckerman/docs/dist" "/Users/paulsheldrake/Documents/work/kanopi/zuckerman/zuckerman/patternlab" 10.20.1.3 -alldirs -mapall=501:20
# VAGRANT-END: 501 e48f4e7d-f212-4fb7-b14f-5d78fb7df132


# VAGRANT-BEGIN: 501 ae60f4e7-ee00-491d-a138-41cb0d902dec
"/Users/paulsheldrake/.vagrant.d/cache/geerlingguy/ubuntu1604" 192.168.21.12 -alldirs -mapall=501:20
"/Users/paulsheldrake/Documents/work/kanopi/sfcm/drupalvm-sfcm" 192.168.21.12 -alldirs -mapall=501:20
"/Users/paulsheldrake/Documents/work/kanopi/sfcm/sfcm" 192.168.21.12 -alldirs -mapall=501:20
# VAGRANT-END: 501 ae60f4e7-ee00-491d-a138-41cb0d902dec





# VAGRANT-BEGIN: 501 8ae7354d-589e-4262-8a03-e964bada3fb4
"/Users/paulsheldrake/.vagrant.d/cache/geerlingguy/ubuntu1604" 192.168.99.11 -alldirs -mapall=501:20
"/Users/paulsheldrake/Documents/work/kanopi/sfpl/code" 192.168.99.11 -alldirs -mapall=501:20
"/Users/paulsheldrake/Documents/work/kanopi/sfpl/drupal-vm" 192.168.99.11 -alldirs -mapall=501:20
# VAGRANT-END: 501 8ae7354d-589e-4262-8a03-e964bada3fb4


# VAGRANT-BEGIN: 501 c2016ecd-8a1f-4f1a-8395-914247b50452
"/Users/paulsheldrake/.vagrant.d/cache/geerlingguy/ubuntu1604" 192.168.82.23 -alldirs -mapall=501:20
"/Users/paulsheldrake/Documents/work/brain1st/brain-1st" 192.168.82.23 -alldirs -mapall=501:20
"/Users/paulsheldrake/Documents/work/brain1st/drupalvm_brain1st" 192.168.82.23 -alldirs -mapall=501:20
# VAGRANT-END: 501 c2016ecd-8a1f-4f1a-8395-914247b50452


# VAGRANT-BEGIN: 501 887771ae-e3e3-404c-a68f-d591e6dac7e8
"/Users/paulsheldrake/.vagrant.d/cache/geerlingguy/ubuntu1604" 192.168.88.54 -alldirs -mapall=501:20
"/Users/paulsheldrake/Documents/work/kanopi/ibew/drupalvm_ibew_46" 192.168.88.54 -alldirs -mapall=501:20
"/Users/paulsheldrake/Documents/work/kanopi/ibew/ibew1269" 192.168.88.54 -alldirs -mapall=501:20
# VAGRANT-END: 501 887771ae-e3e3-404c-a68f-d591e6dac7e8






# VAGRANT-BEGIN: 501 eba2a7d3-3c8e-4b21-a354-31a5e34b0d19
"/Users/paulsheldrake/Documents/work/kanopi/ischool/www-ischool" 10.20.1.2 -alldirs -mapall=501:20
# VAGRANT-END: 501 eba2a7d3-3c8e-4b21-a354-31a5e34b0d19


# VAGRANT-BEGIN: 501 c8452f93-258e-44b3-8151-d851e03f0dcb
"/Users/paulsheldrake/.vagrant.d/cache/geerlingguy/ubuntu1604" 192.168.23.39 -alldirs -mapall=501:20
"/Users/paulsheldrake/Documents/work/kanopi/ndi/drupalvm-ndi-tech" 192.168.23.39 -alldirs -mapall=501:20
"/Users/paulsheldrake/Documents/work/kanopi/ndi/ndiorg" 192.168.23.39 -alldirs -mapall=501:20
# VAGRANT-END: 501 c8452f93-258e-44b3-8151-d851e03f0dcb
# <ds-nfs docksal
/Users/paulsheldrake/docksal 127.0.0.1 192.168.64.100 -alldirs -maproot=0:0
# ds-nfs>

███  FIN
fin version: 1.80.1

███  DOCKER COMPOSE
EXPECTED VERSION: 1.23.1
docker-compose version 1.23.1, build b02f1306
docker-py version: 3.5.0
CPython version: 3.6.6
OpenSSL version: OpenSSL 1.1.0h  27 Mar 2018

███  DOCKER
EXPECTED VERSION: 18.06.1-ce

Client:
 Version:           18.06.1-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        e68fc7a
 Built:             Tue Aug 21 17:21:31 2018
 OS/Arch:           darwin/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.1-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       e68fc7a
  Built:            Tue Aug 21 17:28:38 2018
  OS/Arch:          linux/amd64
  Experimental:     false

███  DOCKER MACHINE
EXPECTED VERSION: 0.15.0
docker-machine version 0.15.0, build b48dc28d

NAME      ACTIVE   DRIVER       STATE     URL                         SWARM   DOCKER        ERRORS
docksal   *        virtualbox   Running   tcp://192.168.64.100:2376           v18.06.1-ce

███  DOCKSAL: PROJECTS
project             STATUS              virtual host        project root

███  DOCKSAL: VIRTUAL HOSTS

███  DOCKER: RUNNING CONTAINERS
CONTAINER ID        IMAGE                     COMMAND                  CREATED             STATUS                   PORTS                                                    NAMES
da7ff7ff1933        docksal/ssh-agent:1.2     "docker-entrypoint.s…"   4 weeks ago         Up 7 minutes (healthy)                                                            docksal-ssh-agent
ccebaee64fc1        docksal/dns:1.1           "docker-entrypoint.s…"   4 weeks ago         Up 7 minutes (healthy)   192.168.64.100:53->53/udp                                docksal-dns
00eafcb5c679        docksal/vhost-proxy:1.4   "docker-entrypoint.s…"   4 weeks ago         Up 7 minutes (healthy)   192.168.64.100:80->80/tcp, 192.168.64.100:443->443/tcp   docksal-vhost-proxy

███  DOCKER: NETWORKS
NETWORK ID          NAME                DRIVER              SCOPE
a742459a8d4b        _default            bridge              local
593aa4211c33        bridge              bridge              local
30ecf1e71b43        host                host                local
f6e97f927454        none                null                local

███  VIRTUALBOX
EXPECTED VERSION: 5.2.20
5.2.2r119230

███  DOCKSAL MOUNTS
192.168.64.1:/Users/paulsheldrake/docksal on /Users/paulsheldrake/docksal type nfs (rw,noatime,nodiratime,vers=3,rsize=65536,wsize=65536,namlen=255,acregmin=1,acregmax=1,acdirmin=1,acdirmax=1,hard,nocto,nolock,proto=tcp,port=2049,timeo=70,retrans=3,sec=sys,local_lock=all,addr=192.168.64.1)

███  HDD Usage
Filesystem                Size      Used Available Use% Mounted on
/dev/sda1                93.2G     37.7G     50.7G  43% /mnt/sda1
```

</details>

### LINKED ISSUE #973: WSL: DNS service won't start, empty IP address
**Description**
Docksal setup fails under WSL due to being unable to start the DNS service.

**Steps to reproduce the issue:**

1. Follow docksal for WSL installation tutorial
2. curl -fsSL get.docksal.io | DOCKER_NATIVE=1 bash
3. (╯°□°）╯︵ ┻━┻

**Describe the results you received:**
Resetting Docksal services...
 ➔ proxy
 ➔ dns
   upstream
invalid argument "" for "--dns" flag:  is not an ip address
See 'docker run --help'.
 ERROR:  Failed starting the DNS service.

**Output of `fin sysinfo`:**

<details>
  <summary>fin sysinfo output</summary>

  ```
OS
WSL Ubuntu 18.04
Linux DESKTOP-C3IEI56 4.4.0-17134-Microsoft #523-Microsoft Mon Dec 31 17:49:00 PST 2018 x86_64 x86_64 x86_64 GNU/Linux

ENVIRONMENT
MODE :
FIN
fin version: 1.80.1

DOCKER COMPOSE
EXPECTED VERSION: 1.23.1
docker-compose version 1.23.1, build b02f1306
docker-py version: 3.5.0
CPython version: 3.6.7
OpenSSL version: OpenSSL 1.1.0f  25 May 2017

DOCKER
EXPECTED VERSION: 18.06.1-ce

Client:
 Version:           18.09.2
 API version:       1.39
 Go version:        go1.10.6
 Git commit:        6247962
 Built:             Sun Feb 10 04:13:47 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.2
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.6
  Git commit:       6247962
  Built:            Sun Feb 10 04:13:06 2019
  OS/Arch:          linux/amd64
  Experimental:     false

DOCKSAL: PROJECTS
project             STATUS              virtual host        project root

DOCKSAL: VIRTUAL HOSTS
No virtual hosts found

DOCKER: RUNNING CONTAINERS
CONTAINER ID        IMAGE                     COMMAND                  CREATED              STATUS                    PORTS                                                    NAMES
8cfbd995036d        docksal/vhost-proxy:1.4   "docker-entrypoint.s…"   About a minute ago   Up 59 seconds (healthy)   192.168.64.100:80->80/tcp, 192.168.64.100:443->443/tcp   docksal-vhost-proxy

DOCKER: NETWORKS
NETWORK ID          NAME                DRIVER              SCOPE
c88b36a5daac        _default            bridge              local
14f887678a1c        bridge              bridge              local
27189d7589a9        host                host                local
03c7f4b8fd3f        none                null                local

HDD Usage
/usr/local/bin/fin: line 5311: wmic: command not found
  ```

</details>

### PR DESCRIPTION
**Pull request by lmakarov**
## Replaces #890

Full WSL support with both, Docker Desktop for Windows and VirtualBox modes.

Closes #851. Closes #865. Closes #973 (verify), Closes #936, Closes #749

## Replaces #960

- Dropped unused vm use cases
  - `vm create`
  - `vm active`
  - `vm ip <IP>`
- Deprecated `fin vm start/stop` in favor of `fin system start/stop`

Closes #949 (verify), 

---

Testing / regressions testing:

- [x] WSL + VirtualBox
- [x] WSL + Docker Desktop
- [x] Babun + VirtualBox
- [x] Babun + Docker Desktop
- [x] Mac + VirtualBox
- [x] Mac + Docker Desktop
- [x] Linux
- [x] Play with Docker

### CODE REVIEW

File: bin/fin [hunk 1]
```diff
@@ -771,19 +806,15 @@ check_docker_running ()
...
-		docker_machine_start
-		if_failed "Could not start Docksal docker machine properly"
-		# re-check status
-		eval $(docker-machine env --shell=bash "$DOCKER_MACHINE_NAME")
-		docker info >/dev/null
-		docker_status=$?
+	if [[ ${docker_status} -eq 255 ]] && ! is_docker_native; then
+		echo-yellow "It looks like '$DEFAULT_MACHINE_NAME' docker machine is not running."
+		_confirm "Start it now?"
+		system_start && return $?
```
Comment (achekulaev) [inline bin/fin]:
```suggestion
		system_start
		return $?
```

File: bin/fin [hunk 2]
```diff
@@ -771,19 +806,15 @@ check_docker_running ()
...
-	if [[ ${docker_status} -eq 255 ]] && (! is_docker_native && ! is_wsl); then
-		echo-yellow "It looks like '$DOCKER_MACHINE_NAME' docker machine is not running."
-		_confirm "Run 'fin vm start' to start it now?"
-		docker_machine_start
-		if_failed "Could not start Docksal docker machine properly"
-		# re-check status
-		eval $(docker-machine env --shell=bash "$DOCKER_MACHINE_NAME")
-		docker info >/dev/null
-		docker_status=$?
+	if [[ ${docker_status} -eq 255 ]] && ! is_docker_native; then
```
Comment (achekulaev) [inline bin/fin]:
```suggestion
	if [[ ${docker_status} == 255 ]] && ! is_docker_native; then
```

File: bin/fin [hunk 3]
```diff
@@ -870,13 +901,19 @@ is_vbox_version ()
 
 is_docker_version ()
 {
+	# Skip the check on Pplay-with-Docker and Katacoda
```
Comment (achekulaev) [inline bin/fin]:
```suggestion
	# Skip the check on Play-with-Docker and Katacoda
```

File: bin/fin [hunk 4]
```diff
@@ -870,13 +901,19 @@ is_vbox_version ()
...
+	(is_pwd || is_katacoda) && return
+
 	! which docker >/dev/null 2>&1 && return 1
 	local version=$(docker version --format '{{.Client.Version}}' 2>/dev/null)
 	[[ $(ver_to_int "$REQUIREMENTS_DOCKER") -le $(ver_to_int "$version") ]]
 }
 
 is_docker_server_version ()
 {
+	# Skip the check on Pplay-with-Docker and Katacoda
```
Comment (achekulaev) [inline bin/fin]:
```suggestion
	# Skip the check on Play-with-Docker and Katacoda
```

File: bin/fin [hunk 5]
```diff
@@ -2501,13 +2493,18 @@ smb_share_add()
 # Removes Docksal SMB shares
 smb_share_remove ()
 {
-	local drives=$(wmic logicaldisk get drivetype,name | grep 3 | awk '{print $2}' | sed 's/\r//g')
+	local drives
+	if is_wsl; then drives=$(ls /mnt); else drives=$(ls /cygpath); fi
```
Comment (achekulaev) [inline bin/fin]: What if `/mnt` contains unrelated mounts?

File: bin/fin [hunk 6]
```diff
@@ -2608,98 +2610,54 @@ vm ()
...
-			fi
-			;;
 		start)
-			if is_docker_machine_running; then
-				echo "Machine \"$DOCKER_MACHINE_NAME\" is already running."
-			else
-				docker_machine_start
-			fi
+			# Show a message and pause for input.
+			echo -e "Deprecated. Please use ${yellow}fin system start${NC} instead."; read -p "Press any key to continue..."
```
Comment (achekulaev) [inline bin/fin]:
```suggestion
			echo -e "Deprecated. Please use ${yellow}fin system start${NC} instead."
			read -p "Press any key to continue..."
```

File: bin/fin [hunk 7]
```diff
@@ -3553,19 +3525,33 @@ detect_dns ()
...
+					DOCKSAL_DNS_UPSTREAM="${dns}"
+					break
+				fi
+			done
 		fi
 	fi
 
-	# If localhost address was detected as upstream, fall back to the default
-	if [[ "$DOCKSAL_DNS_UPSTREAM" =~ "127.0."* ]]; then
+	# If localhost address or empty was detected as upstream, fall back to the default
```
Comment (achekulaev) [inline bin/fin]:
```suggestion
	# If detected upstream points to localhost or is empty, then fall back to the default
```

File: bin/fin [hunk 8]
```diff
@@ -4325,49 +4368,32 @@ update ()
...
 			fi
 		fi
 
-		if ! is_linux && ! is_docker_native && ! is_wsl && which 'docker-machine' >/dev/null 2>&1; then
+		if ! is_linux && ! is_docker_native && which 'docker-machine' >/dev/null 2>&1; then
 			if is_docker_machine_exist && ! is_docker_machine_running; then
-				echo-error "Docker Machine '$DOCKER_MACHINE_NAME' should be running to perform update" \
-					"Start it with ${yellow}fin vm start${NC} or destroy it with ${yellow}fin vm remove${NC} if you want it re-created."
+				echo-error "Docker Machine '$DEFAULT_MACHINE_NAME' should be running to perform update" \
+					"Start it with ${yellow}fin system start${NC} or destroy it with ${yellow}fin vm kill${NC} if you want it re-created."
```
Comment (achekulaev) [inline bin/fin]:
```suggestion
					"Start it with ${yellow}fin system start${NC} or destroy it with ${yellow}fin vm remove${NC} if you want it re-created."
```

`docker-machine kill` does not remove the machine

File: bin/fin (hunk 5 above)
Comment (lmakarov) [reply bin/fin]: I was thinking about that and considered that as an edge case that can be addressed if that ever happens. I like how simple and clean it is now, but if you feel a the more complex treatment via using `wmic.exe` is better, I'm fine with that too. Leave/revert?

Comment (lmakarov) [reply bin/fin]: Also, `ls /cygpath` was a typo, should be `ls /cygdrive`. Already updated that.

Comment (lmakarov) [reply bin/fin]: And here's that edge case - https://github.com/docksal/docksal/issues/1059. So, going back to `wmic.exe`

**Pull request merged**